src/Compilers/CSharp/Test/Symbol/SymbolDisplay/SymbolDisplayTests.cs

```diff
@@ -7652,5 +7652,171 @@ public void RecordDeclaration()
+@"
+Imports N
+Public Class Test
+    Inherits A(Of Object)
+End Class";
+            var compCS = CreateCompilation(new[] { sourceCS, MaybeNullAttributeDefinition });
+            var refCS = compCS.EmitToImageReference();
+            var compVB = CreateVisualBasicCompilation(GetUniqueName(), sourceVB, referencedAssemblies: new[] { MscorlibRef, refCS });
+            var member = compVB.GetTypeByMetadataName("Test").BaseType.GetMember("P");
+            var format = new SymbolDisplayFormat(
+                memberOptions: SymbolDisplayMemberOptions.IncludeParameters | SymbolDisplayMemberOptions.IncludeType | SymbolDisplayMemberOptions.IncludeModifiers,
+                genericsOptions: SymbolDisplayGenericsOptions.IncludeTypeParameters,
+                miscellaneousOptions: SymbolDisplayMiscellaneousOptions.IncludeNullableReferenceTypeModifier);
+
+            Verify(
+                member.ToDisplayParts(format),
+                "Overloads P As Object",
+                SymbolDisplayPartKind.Keyword,
+                SymbolDisplayPartKind.Space,
+                SymbolDisplayPartKind.PropertyName,
+                SymbolDisplayPartKind.Space,
+                SymbolDisplayPartKind.Keyword,
+                SymbolDisplayPartKind.Space,
+                SymbolDisplayPartKind.ClassName);
+        }
+
+        [WorkItem(48023, "https://github.com/dotnet/roslyn/issues/48023")]
+        [Fact]
+        public void NullableAttributes_CSharpReferencesVisualBasic()
+        {
+            var sourceVB =
+@"
+Namespace System.Diagnostics.CodeAnalysis
+    <AttributeUsage(AttributeTargets.Field Or AttributeTargets.Parameter Or AttributeTargets.[Property] Or AttributeTargets.ReturnValue)>
+    Public NotInheritable Class MaybeNullAttribute
+        Inherits Attribute
+    End Class
+End Namespace
+
+Namespace N
+    Public Class A(Of T)
+        <System.Diagnostics.CodeAnalysis.MaybeNull>
+        Public ReadOnly Property P As T
+    End Class
+End Namespace
+";
+            var sourceCS =
+@"
+#nullable enable
+
+namespace N
+{
+    public class Test: A<object> { }
+}
+";
+            var compVB = CreateVisualBasicCompilation(GetUniqueName(), sourceVB, referencedAssemblies: new[] { MscorlibRef });
+            var refVB = compVB.EmitToImageReference();
+            var compCS = CreateCompilation(sourceCS, references: new[] { refVB });
+            var member = compCS.GetTypeByMetadataName("N.Test").BaseType().GetMember("P");
+            var format = new SymbolDisplayFormat(
+                memberOptions: SymbolDisplayMemberOptions.IncludeParameters | SymbolDisplayMemberOptions.IncludeType | SymbolDisplayMemberOptions.IncludeModifiers,
+                genericsOptions: SymbolDisplayGenericsOptions.IncludeTypeParameters,
+                miscellaneousOptions: SymbolDisplayMiscellaneousOptions.IncludeNullableReferenceTypeModifier);
+
+            Verify(
+                member.ToDisplayParts(format),
+                "Object P",
```

Comment:
I would have expected to get `Object? P` here, given that VB defines and applies the expected attribute (given that it doesn't matter where `MaybeNullAttribute` is defined).

Reply:
According to the [nullable-metadata](https://github.com/dotnet/roslyn/blob/master/docs/features/nullable-metadata.md) documentation, this scenario (VB defines NRT annotations for C#) is not supported. I think I should therefore simplify the test, to avoid confusion.

Reply:
I removed the nullable attributes in the VB code in 60553d24.